Scripts/SL-MsdFileParser.lua

```diff
@@ -16,37 +16,33 @@
 -- we'll recover.
 
 function ParseMsdFile(steps)
-	local function AddParam(t, p, plen)
+	local function AddParam(t, p, plen, fileType)
 		-- table.concat(table_name, separator, start, end)
 		local param = table.concat(p, '', 1, plen)
 
 		-- -- Normalize all line endings to \n and remove all leading and trailing whitespace.
-		param = param:gsub('\r\n', '\n'):gsub('\r', '\n'):match('^%s*(.-)%s*$')
+		param = param:gsub('\r\n?', '\n'):match('^%s*(.-)%s*$')
 
 		-- Field specific modifications. We length check the last table to make sure we're
 		-- actually parsing what we want.
 		-- TODO(teejusb): We should probably do this for most fields for consistency.
-		-- NOTE(teejusb): This is for *.sm files only. This is easily changeable,
-		-- I just haven't got around to it yet.
-		if((#t[#t] == 6 and t[#t][1] == 'NOTES')) then
+		if #t[#t] == 6 and (t[#t][1] == 'NOTES' or t[#t][1] == 'NOTES2') and fileType == "sm" then
 			-- Spaces don't matter for the chart data itself, remove them all.
+			-- NOTE(teejusb): NOTES in the SM file has multiple parts. We specifically only want to strip
+			-- the 6th part (the chart data), as the others can contain spaces e.g. chart description.
 			param = param:gsub(' ', '')
-		elseif((#t[#t] == 1 and t[#t][1] == 'BPMS')) then
+		elseif #t[#t] == 1 and (t[#t][1] == 'NOTES' or t[#t][1] == 'NOTES2') and fileType == "ssc" then
+			-- Spaces don't matter for the chart data itself, remove them all.
+			-- NOTE(teejusb): NOTES in an SSC file only contains the chart data itself.
+			param = param:gsub(' ', '')
+		elseif #t[#t] == 1 and t[#t][1] == 'BPMS' then
 			-- Line endings and spaces don't matter for BPMs, remove them all.
-			param = param:gsub('\n', ''):gsub(' ', '')
+			param = param:gsub('\n? ?', '')
```

Comment:
Maybe %s here too?

Reply:
%s might work here since we don't care about new lines for BPMS, but again I'll need to verify with groovestats.

Reply:
Resolved!